packages/redis-message-deduplication-store/lib/RedisMessageDeduplicationStore.ts

```
@@ -0,0 +1,40 @@
+import type { MessageDeduplicationStore } from '@message-queue-toolkit/core'
+import type { Redis } from 'ioredis'
+
+export type RedisMessageDeduplicationStoreDependencies = {
+  redis: Redis
+}
+
+export type RedisMessageDeduplicationStoreConfig = {
+  keyPrefix?: string
+}
+
+export class RedisMessageDeduplicationStore implements MessageDeduplicationStore {
+  private readonly redis: Redis
+  private readonly config: RedisMessageDeduplicationStoreConfig
+
+  constructor(
+    dependencies: RedisMessageDeduplicationStoreDependencies,
+    config: RedisMessageDeduplicationStoreConfig,
+  ) {
+    this.redis = dependencies.redis
+    this.config = config
+  }
+
+  async setIfNotExists(key: string, value: string, ttlSeconds: number): Promise<boolean> {
+    const keyWithPrefix = this.getKeyWithOptionalPrefix(key)
+    const result = await this.redis.set(keyWithPrefix, value, 'EX', ttlSeconds, 'NX')
```

what if we error out here? Wonder if we may want to try-catch and resolve to false in that case, or fail fast and break the flow

---

If we want to handle this specific case, should I handle all of cases when we're communicating with redis to gracefully handle connection errors?

---

@kibertoad I've added handling of dedup-related errors on `AbstractQueueService` level. In case of an error, it is handled by `handleError` (method that was already in place, it logs the errors and reports it) and the message is being processed with deduplication turned off. Let me know if that's a good approach to you.